plugins/org.eclipse.fordiac.ide.util/src/org/eclipse/fordiac/util/marker/UtilityMarkerHelper.java

```diff
@@ -0,0 +1,123 @@
+ *
+ * Contributors:
+ *   Mario Kastner - initial implementation and/or documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.util.marker;
+
+import java.util.stream.Stream;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.fordiac.ide.model.errormarker.ErrorMarkerBuilder;
+import org.eclipse.fordiac.ide.model.errormarker.FordiacErrorMarker;
+import org.eclipse.fordiac.ide.model.libraryElement.AutomationSystem;
+import org.eclipse.fordiac.ide.model.libraryElement.LibraryElement;
+import org.eclipse.fordiac.ide.ui.FordiacLogHelper;
+import org.eclipse.fordiac.ide.ui.editors.EditorUtils;
+
+public class UtilityMarkerHelper {
+
+	public static EObject getMarkedElement(final MarkerDescriptor markerDescriptor, final IResource resource) {
+		try {
+			return Stream.of(resource.findMarkers(markerDescriptor.ID(), true, IResource.DEPTH_ZERO))
+					.map(UtilityMarkerHelper::getTarget).findAny().orElse(null);
+		} catch (final CoreException e) {
+			FordiacLogHelper.logError("Error accessing marked element", e); //$NON-NLS-1$
+		}
+		return null;
+	}
+
+	public static IMarker getMarker(final MarkerDescriptor markerDescriptor) {
+		if (EditorUtils.getCurrentActiveEditor()
+				.getAdapter(LibraryElement.class) instanceof final AutomationSystem sys) {
+			try {
+				return Stream
+						.of(sys.getTypeEntry().getFile().findMarkers(markerDescriptor.ID(), true, IResource.DEPTH_ZERO))
+						.findAny().orElse(null);
+			} catch (final CoreException e) {
+				FordiacLogHelper.logError("Error accessing marked element", e); //$NON-NLS-1$
+			}
+		}
+		return null;
+	}
+
+	public static void deleteElementMarker(final MarkerDescriptor markerDescriptor, final IResource resource) {
+		deleteMarker(markerDescriptor.ID(), resource);
+	}
+
+	public static void deleteElementMarker(final String id) {
+		if ((EditorUtils.getCurrentActiveEditor()
+				.getAdapter(LibraryElement.class) instanceof final AutomationSystem sys)
+				&& (sys.getTypeEntry().getFile() instanceof final IResource resource)) {
+			deleteMarker(id, resource);
+		}
+	}
+
+	private static void deleteMarker(final String id, final IResource resource) {
+		try {
+			resource.deleteMarkers(id, false, IResource.DEPTH_ZERO);
+		} catch (final CoreException e) {
+			FordiacLogHelper.logError("Error deleting marker", e); //$NON-NLS-1$
+		}
+	}
+
+	public static void addElementMarker(final MarkerDescriptor markerDescriptor, final IResource resource,
+			final EObject target) {
+		try {
+			ErrorMarkerBuilder.createErrorMarkerBuilder(markerDescriptor.name()).setType(markerDescriptor.ID())
+					.setTarget(target).setSeverity(IMarker.SEVERITY_INFO).createMarker(resource);
+		} catch (final CoreException e) {
+			FordiacLogHelper.logError("Cannot add marker", e); //$NON-NLS-1$
+		}
```

Comment:
As there may only be once such marker at a given time, I would suggest updating the marker with:
```suggestion
		FordiacMarkerHelper
				.updateMarkers(resource, markerDescriptor.ID(),
						List.of(ErrorMarkerBuilder.createErrorMarkerBuilder(markerDescriptor.name())
								.setType(markerDescriptor.ID()).setTarget(target).setSeverity(IMarker.SEVERITY_INFO)),
						true);
```
This already handles deleting any old markers with the given type.